### PR TITLE
[SolidJS + TailwindCSS] Fix:  storybook, tests library and kit config

### LINKED ISSUE #1235: [Bug]: Solid.js + Tailwind goes into white page when starting the app
### What kit and version are you using?

solidjs-tailwind@0.1.0 

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### Actual Behavior

When running `npm install` and then trying to start the kit with `npm run dev`, gives me a white page with the following error:

`caught TypeError: _tmpl$ is not a function`

### Expected Behavior

I should be able to run the kit with the latest version of `npm` or any package manager and `node`

### Reproduction

Run the commands: `npm install` then `npm run dev`

The kit won't start, with an error in the console saying: 
`caught TypeError: _tmpl$ is not a function`

### Additional Information

_No response_

### LINKED ISSUE #1236: [Feature]: Solid.js + Tailwind kit deprecated library
### Is your feature request related to a problem? Please describe.

The library `solid-testing-library` is desecrated and its use it is not recommended anymore

### Describe the solution you'd like

The package should replace the library with: `@solidjs/testing-library`

### Describe alternatives you've considered

Leave a note in the README.md file in case this change is not possible.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jesus4497**
## Type of change

- [ ] Documentation change
- [x] Bug fix

## Summary of change

Storybook and the kit itself were broken, so I made the following changes:

- Upgraded dependencies
  - Upgraded storybook from 6.5 to 7
    - Updated config files
    - Updated Eslint file
- Upgraded SolidJS and Vite dependencies
- Changed deprecated dependency `solid-testing-library` to `@solidjs/testing-library`

## Checklist

- [x] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [x] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [x] This fix resolves #1235 and resolves #1236
- [x] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/solidjs-tailwind/src/components/CounterExample/Counter.spec.jsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { fireEvent, render, screen } from 'solid-testing-library';
+import { fireEvent, render, screen } from '@solidjs/testing-library';
```
Comment (jesus4497) [inline starters/solidjs-tailwind/src/components/CounterExample/Counter.spec.jsx:1]: Testing-library [recommends](https://testing-library.com/docs/user-event/intro) using the [user-event](https://github.com/testing-library/user-event) module for most user-interactions and falling back to fireEvent for things user-event can't do - we should do a follow up PR for this

**Review by jesus4497 — COMMENTED** [review]
Also, we might want to do another follow up PR to make the kit with TypeScript

File: starters/solidjs-tailwind/.gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ node_modules
 dist
 
 # .vscode
+.vscode
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/.gitignore:5]: is this a duplicate?

File: starters/solidjs-tailwind/package.json [hunk 3]
```diff
@@ -2,18 +2,8 @@
...
-    "Maarten Bicknese <maarten.bicknese@thisdot.co>",
-    "Chinedu Okpala <chinedu.okpala@thisdot.co>",
-    "Daian Scuarissi <daian@thisdot.co>",
-    "Mattia Magi <mattia.magi@thisdot.co>",
-    "Mark Shenouda <mark@thisdot.co>",
-    "Victor Umeh <victor.umeh@thisdot.co>",
-    "Dane Grant <dane@thisdot.co>",
-    "Jan Kaiser <jan.kaiser@thisdot.co>",
-    "Jerry Hogan <jerry.hogan@thisdot.co>"
-  ],
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:16]: are we sure we want to remove this?
I vaguely remember a discussion a while back about adding contributors in Slack.
So I would check on this

File: starters/solidjs-tailwind/package.json [hunk 4]
```diff
@@ -23,47 +13,53 @@
...
-    "@storybook/addon-measure": "6.5.16",
-    "@storybook/addon-outline": "6.5.16",
-    "@storybook/builder-vite": "0.4.2",
-    "@storybook/client-api": "6.5.16",
-    "@storybook/client-logger": "6.5.16",
-    "@storybook/core-common": "6.5.16",
-    "@storybook/html": "6.5.16",
+    "@babel/core": "7.21.8",
+    "@solidjs/router": "0.8.2",
+    "@solidjs/testing-library": "^0.7.0",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:23]: this dependency needs to be pinned

File: starters/solidjs-tailwind/package.json [hunk 5]
```diff
@@ -23,47 +13,53 @@
...
+    "@storybook/addon-actions": "7.0.8",
+    "@storybook/addon-backgrounds": "7.0.8",
+    "@storybook/addon-docs": "7.0.8",
+    "@storybook/addon-essentials": "7.0.8",
+    "@storybook/addon-interactions": "^7.0.8",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:28]: this dependency needs to be pinned

File: starters/solidjs-tailwind/package.json [hunk 6]
```diff
@@ -23,47 +13,53 @@
...
+    "@storybook/addon-links": "7.0.8",
+    "@storybook/addon-measure": "7.0.8",
+    "@storybook/addon-outline": "7.0.8",
+    "@storybook/blocks": "^7.0.8",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:32]: this dependency needs to be pinned

File: starters/solidjs-tailwind/package.json [hunk 7]
```diff
@@ -23,47 +13,53 @@
...
+    "@storybook/client-api": "7.0.8",
+    "@storybook/client-logger": "7.0.8",
+    "@storybook/core-common": "7.0.8",
+    "@storybook/testing-library": "^0.0.14-next.2",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:36]: this dependency needs to be pinned

File: starters/solidjs-tailwind/package.json [hunk 8]
```diff
@@ -23,47 +13,53 @@
...
     "@testing-library/jest-dom": "5.16.5",
-    "@typescript-eslint/parser": "5.52.0",
-    "autoprefixer": "10.4.13",
-    "eslint": "8.34.0",
-    "eslint-plugin-solid": "0.9.4",
+    "@typescript-eslint/parser": "5.59.2",
+    "autoprefixer": "10.4.14",
+    "eslint": "8.39.0",
+    "eslint-plugin-solid": "0.12.1",
+    "eslint-plugin-storybook": "^0.6.12",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:42]: this dependency needs to be pinned

File: starters/solidjs-tailwind/package.json [hunk 9]
```diff
@@ -23,47 +13,53 @@
...
-    "vite": "4.1.4",
-    "vite-plugin-solid": "2.5.0",
-    "vitest": "0.28.5",
+    "jsdom": "22.0.0",
+    "msw": "1.2.1",
+    "postcss": "8.4.23",
+    "prettier": "2.8.8",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "storybook": "^7.0.8",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:50]:
```suggestion
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "storybook": "7.0.8",
```

File: starters/solidjs-tailwind/package.json [hunk 10]
```diff
@@ -23,47 +13,53 @@
...
+    "storybook-addon-mock": "4.0.0",
+    "storybook-solidjs": "^1.0.0-beta.2",
+    "storybook-solidjs-vite": "^1.0.0-beta.2",
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/package.json:53]:
```suggestion
    "storybook-solidjs": "1.0.0-beta.2",
    "storybook-solidjs-vite": "1.0.0-beta.2",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Looks good 👍 

there were just a couple of places where the dependencies needed to be set to the fixed values.
And I left a couple of questions

File: starters/solidjs-tailwind/package.json (hunk 3 above)
Comment (jesus4497) [reply starters/solidjs-tailwind/package.json]: oh true!  I deleted this accidentally. Added myself muahahah

File: starters/solidjs-tailwind/src/components/CounterExample/CounterExample.stories.jsx [hunk 11]
```diff
@@ -2,9 +2,10 @@ import { CounterExample } from '.';
...
 export default {
   title: 'Example/ Counter Example',
+  component: CounterExample,
   argTypes: {},
 };
 
 const Template = (args) => <CounterExample {...args} />;
 
-export const Default = Template.bind({});
+export const CounterExamle = Template.bind({});
```
Comment (jdwilkin4) [inline starters/solidjs-tailwind/src/components/CounterExample/CounterExample.stories.jsx:11]: one small thing. spelling error
```suggestion
export const CounterExample = Template.bind({});
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
One small spelling error fix but then we should be good to go

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**